ledger-core/virtual/integration/deactivate_test.go

```diff
@@ -746,3 +746,165 @@ func TestVirtual_CallMethod_After_Deactivation(t *testing.T) {
 
 	mc.Finish()
 }
+
+func TestVirtual_DeactivateObject_FinishPartialDeactivation(t *testing.T) {
+	insrail.LogCase(t, "C5508")
+	const (
+		origMem     = "object orig mem"
+		checkMethod = "Check"
+	)
+	defer commonTestUtils.LeakTester(t)
+
+	cases := []struct {
+		name      string
+		isolation contract.MethodIsolation
+	}{
+		{
+			name:      "Tolerable-Dirty",
+			isolation: tolerableFlags(),
+		},
+		{
+			name:      "Intolerable-Validated",
+			isolation: intolerableFlags(),
+		},
+	}
+	for _, testCase := range cases {
+		t.Run(testCase.name, func(t *testing.T) {
+			mc := minimock.NewController(t)
+
+			server, ctx := utils.NewUninitializedServer(nil, t)
+
+			oneExecutionEnded := server.Journal.WaitStopOf(&execute.SMExecute{}, 1)
+
+			runnerMock := logicless.NewServiceMock(ctx, mc, func(execution execution.Context) string {
+				return execution.Request.CallSiteMethod
+			})
+			server.ReplaceRunner(runnerMock)
+			server.Init(ctx)
+
+			var (
+				class               = gen.UniqueGlobalRef()
+				objectRef           = reference.NewSelf(server.RandomLocalWithPulse())
+				p1                  = server.GetPulse().PulseNumber
+				deactivateIsolation = contract.MethodIsolation{
+					Interference: contract.CallTolerable,
+					State:        contract.CallDirty,
+				}
+				stateRef = server.RandomGlobalWithPulse()
+			)
+
+			server.IncrementPulseAndWaitIdle(ctx)
+			outgoing := server.BuildRandomOutgoingWithPulse()
+			checkOutgoing := server.BuildRandomOutgoingWithPulse()
+
+			{
+				runnerMock.AddExecutionClassify(checkMethod, testCase.isolation, nil)
+				if testCase.isolation.State == contract.CallValidated {
+					runnerMock.AddExecutionMock(checkMethod).AddStart(
+						nil,
```

Comment:
check that it's validated memory, please

Reply:
done